Append APP_ to METADATA_DIR_PATH to make it clear it's for the app

mokagio opened the pull request:
### Description

That is, so we won't confuse it with the metadata Fastlane uses. See https://github.com/woocommerce/woocommerce-android/pull/6814#discussion_r907027309.

We now have `APP_METADATA_DIR_PATH` for `WooCommerce/metadata` and `PLAY_STORE_METADATA_DIR_PATH` for `fastlane/metadata/android`. The second is used only once, but will be used again once I work on using `supply` to upload the release notes.

### Testing instructions

This PR touches four lanes. I tested three of them and pushed the result to make sure the paths didn't change:

- `update_appstore_strings` https://github.com/woocommerce/woocommerce-android/pull/6908/commits/82fe8e8b54a990aa02db4ddb659a603f1adfeb08
- `download_metadata_strings` https://github.com/woocommerce/woocommerce-android/pull/6908/commits/9667c0e30ffa2588d154116f06855d40816a759a
- `download_promo_strings` https://github.com/woocommerce/woocommerce-android/pull/6908/commits/329d3c6b0f8517b59fe6d6a6b5e0a2f1a7ad9790 – This creates a lot of new files, but it's because those were never tracked in the first place. Also notice how the `en-US` subfolder has the right values coming from `WooCommerce/metadata`.

The fourth is `create_promo_screenshots` which I didn't run because 1) I don't have screenshots locally, 2) it would have taken a lot of time for something that can be verified by reading the code, and 3) I noticed a possible issue with the path used which, if correct, would invalidate the manual test.

---

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. – It does not

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

mokagio committed:
Append `APP_` to `METADATA_DIR_PATH` to make it clear it's for the app

That is, so we won't confuse it with the metadata Fastlane uses.

See
https://github.com/woocommerce/woocommerce-android/pull/6814#discussion_r907027309

mokagio committed:
Extract Play Store metadata path in constant in `Fastfile`

This is related to the rename from the previous commit,
b75e4fe942415af6897de9352a18b2aca6b96c79, and will come handy soon once
we'll look into that path to upload its content to the Play Store using
`upload_to_play_store`.

mokagio commented on fastlane/Fastfile:
```diff
@@ -706,7 +705,7 @@ platform :android do
       "promo_screenshot_5.txt",
     ].each do |filename|
-      source = File.join(METADATA_DIR_PATH, filename)
+      source = File.join(APP_METADATA_DIR_PATH, filename)
```
I was surprised in seeing this path used here, but it is correct in the context of the how the codebase is structured at the moment. See https://github.com/woocommerce/woocommerce-android/tree/07787d2e86bef83fb58ae886a5295a292638d2e7/WooCommerce/metadata:

<img width="1308" alt="image" src="https://user-images.githubusercontent.com/1218433/178186800-0f42e21e-aaac-47c4-b6e9-70c40ebf967d.png">

AliSoftware replied:
Yeah the way our metadata originals vs final files are organized is so confusing imho. And that's without mentioning folders like `fastlane/playstoreres` which are yet another source for those… how confusing. 😒 
But that kind of cleanup is for another PR 🤷

mokagio commented on fastlane/Fastfile:
```diff
@@ -766,7 +765,7 @@ platform :android do
     promo_screenshots(
       orig_folder: RAW_SCREENSHOTS_PROCESSING_DIR,
-      metadata_folder: METADATA_DIR_PATH,
+      metadata_folder: APP_METADATA_DIR_PATH,
```
I don't think `METADATA_DIR_PATH` (`WooCommerce/metadata`) was the correct value here.

In WordPress Android, we [use a constant](https://github.com/wordpress-mobile/WordPress-Android/blob/a3285a9d49ac75440612f8537283c54f1ed4b096/fastlane/lanes/screenshots.rb#L294) that [resolves to `fastlane/metadata/android`](https://github.com/wordpress-mobile/WordPress-Android/blob/a3285a9d49ac75440612f8537283c54f1ed4b096/fastlane/lanes/screenshots.rb#L4)

I'll need to followup on this with the author and possibly open a PR for this.

AliSoftware replied:
I agree it's probably not the correct value, especially since `WooCommerce/metadata` only contains originals, and doesn't contain any subfolder per locale containing the translations.

On the other hand, the code from the `download_promo_strings` lane above [downloads the translations from GlotPress into `fastlane/playstoreres/metadata`](https://github.com/woocommerce/woocommerce-android/blob/53029c6fb27795776564e4ced21185bd8f5144cc/fastlane/Fastfile#L691-L695) (as if things were not confusing enough).

So I think at least as a first quick-fix, the right value for this one should be changed to `fastlane/playstoreres/metadata`, to match what we currently have, however confusing it is.

And then, in a subsequent PR, it would be nice to do a lot of cleanup about all that, to avoid having 3 sources of truth (`WooCommerce/metadata`, `fastlane/metadata` and `fastlane/appstoreres/metadata`) and instead only have 2 (originals in either `WooCommerce/metadata` …or maybe `fastlane/metadata/default`?, and final translations in `fastlane/metadata`). But that more involved cleanup will be for a follow-up PR.

See also paaHJt-3sE-p2#comment-6146 on the fact that some of our projects have two lanes to download metadata (one for store-related metadata, one for screenshot-related ones), and the suggestion I gave about how to clean it up back then.

mokagio replied:
I followed up on this here: https://github.com/woocommerce/woocommerce-android/pull/6919

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -16,9 +16,11 @@ end
 INSTALLABLE_BUILD_DOMAIN = 'https://d2twmm2nzpx3bg.cloudfront.net'
-METADATA_DIR_PATH = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'metadata')
-RELEASE_NOTES_PATH = File.join(METADATA_DIR_PATH, 'release_notes.txt')
+APP_METADATA_DIR_PATH = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'metadata')
```
When I read "App Metadata Dir Path" I didn't think of it as "oh, that's the subdir inside the _app_ dir" but instead "that is metadata about the app" — which `fastlane/metadata/*` also is. So I'm not sure if that naming clarifies things enough…

Not sure if I have much better suggestions though. Naming is hard.

wdyt about `SOURCE_METADATA_DIR_PATH` or `METADATA_ORIGINALS_DIR_PATH` to suggest that it's the path to the originals before they are ready — i.e. the original before they are reworded (e.g. Editorialization of Release Notes) and that we send them for translation via the `.po` file — as opposed to the final metadata that will be really sent to PlayStore by `supply` in the end?

mokagio replied:
`METADATA_ORIGINALS_DIR_PATH` "the original before they are reworded" sounds good, I'll update to that one.

I might make it `ORIGINAL_METADATA_DIR_PATH`, though, so it matches `PLAY_STORE_...`.

mokagio committed:
Rename `APP_METADATA_DIR_PATH` to `ORIGINAL_METADATA_DIR_PATH`

See suggestion at
https://github.com/woocommerce/woocommerce-android/pull/6907#discussion_r918311648

The rational is that those metadata are not really used "in the app".

AliSoftware commented on fastlane/Fastfile:
```diff
@@ -16,9 +16,11 @@ end
 INSTALLABLE_BUILD_DOMAIN = 'https://d2twmm2nzpx3bg.cloudfront.net'
-METADATA_DIR_PATH = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'metadata')
-RELEASE_NOTES_PATH = File.join(METADATA_DIR_PATH, 'release_notes.txt')
+ORIGINAL_METADATA_DIR_PATH = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'metadata')
```
Hello, it's me, the nitpicker in chief again! 😂 😅 

I feel like "Original Metadata Dir Path" reads as "The path for the original metadata dir", a bit as in "the one true metadata dir, the original one, not a copy", as opposed to "The path to the metadata dir containing the originals"

While `ORIGINALS_METADATA_DIR_PATH` have more chance of reading as "The path for the originals' metadata dir" as in "the metadata dir where the original copies are" 🤔 

wdyt?

```suggestion
ORIGINALS_METADATA_DIR_PATH = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'metadata')
```

AliSoftware reviewed (approved):
Approving to unblock and because I am mostly OK with that rename, but still left a small nitpick/suggestion on the naming, so I disabled auto-merge to give you the chance to address if it you agree with my suggestion.

mokagio committed:
Use `ORIGINALS` instead of `ORIGINAL` in constant name

This is to avoid confusing the path as being "the original path before
change", which would be incorrect. It is the path for the original
metadata to use as source for translations.

See discussion at
https://github.com/woocommerce/woocommerce-android/pull/6907#discussion_r918870140
